### ISSUE TITLE
False positive in derive_trait_impl_removed

### ISSUE THREAD

**Issue by loewenheim** [obi1kenobi/cargo-semver-checks#499]
### Steps to reproduce the bug with the above code

1. Clone https://github.com/getsentry/rust-sourcemap/
2. Run `cargo semver-checks check-release`

### Actual Behaviour

```
❯ cargo semver-checks check-release
    Updating index
     Parsing sourcemap v6.2.3 (current)
     Parsing sourcemap v6.2.3 (baseline, cached)
    Checking sourcemap v6.2.3 -> v6.2.3 (no change)
   Completed [   0.052s] 45 checks; 44 passed, 1 failed, 0 unnecessary

--- failure derive_trait_impl_removed: built-in derived trait no longer implemented ---

Description:
A public type has stopped deriving one or more traits. This can break downstream code that depends on those types implementing those traits.
        ref: https://doc.rust-lang.org/reference/attributes/derive.html#derive
       impl: https://github.com/obi1kenobi/cargo-semver-check/tree/v0.22.1/src/lints/derive_trait_impl_removed.ron

Failed in:
  type RewriteOptions no longer derives Clone, in /Users/sebastian/code/rust-sourcemap/src/types.rs:22
  type RewriteOptions no longer derives Debug, in /Users/sebastian/code/rust-sourcemap/src/types.rs:22
  type RewriteOptions no longer derives Copy, in /Users/sebastian/code/rust-sourcemap/src/types.rs:22
       Final [   0.054s] semver requires new major version: 1 major and 0 minor checks failed
```

### Expected Behaviour

No change should be reported. The type in question, `RewriteOptions`, has never implemented `Clone`, `Copy`, or `Debug`.

The result of running the command on the `6.2.3` tag is the same.

#### Software version

cargo-semver-checks 0.22.1

#### Operating system

macOS 13.2 (Darwin 22.3.0)

#### Command-line

```bash
/Users/sebastian/.cargo/bin/cargo-semver-checks semver-checks --bugreport 
```

#### cargo version

```
> cargo -V 
cargo 1.71.0 (cfd3bbd8f 2023-06-08)
```

#### Compile time information

- Profile: release
- Target triple: x86_64-apple-darwin
- Family: unix
- OS: macos
- Architecture: x86_64
- Pointer width: 64
- Endian: little
- CPU features: cmpxchg16b,fxsr,sse,sse2,sse3,ssse3
- Host: x86_64-apple-darwin

**Comment by obi1kenobi** [comment]
Thanks for the report, that does look strange.

I don't seem to be able to reproduce it the steps provided, nor with any of the surrounding tags:
```
$ git status
HEAD detached at 6.2.3
nothing to commit, working tree clean
predrag@Dragon:/tmp/rust-sourcemap$ cargo semver-checks check-release
    Updating index
     Parsing sourcemap v6.2.3 (current)
     Parsing sourcemap v6.2.3 (baseline, cached)
    Checking sourcemap v6.2.3 -> v6.2.3 (no change)
   Completed [   0.043s] 45 checks; 45 passed, 0 unnecessary
```
I also tried tag `6.3.0`, the current `master` branch (`34ef3574b1fa99e66cdc88b31142404d730ed207`), as well as the most recent commit with your name (`70c51e4e4d87e21380ca39c11251f0d162e69ef7`). All checks passed every time.

Two things we can do, and I think we should do both:
- Could you let me know what commit hash in that repository reproduces this error?
- Inside your build `target` directory, there should be a file at path `semver-checks/cache/registry-sourcemap-6_2_3.json`. Can you attach it here? It's the cached rustdoc JSON of the project being used as baseline, and that's where cargo-semver-checks claims it saw `Clone, Copy, Debug` impls for the affected type. I'd love to take a look at that file directly and compare it to what I'm seeing.

Sorry for the inconvenience and thanks for the help!

**Comment by loewenheim** [comment]
Indeed, in the source code in that directory, the `derive` annotation exists. That is very weird. I dimly recall locally adding them as a test, and somehow the registry must have gotten confused.

**Comment by obi1kenobi** [comment]
Whew. If wiping that directory from the registry and re-running cargo-semver-checks stops triggering this problem, I'm inclined to call it resolved for now. If I understand correctly, it's possible to edit source code in the registry directory and cargo won't catch or prevent that, likely by design.

**Comment by loewenheim** [comment]
Yup, after removing the directory from the registry and the `json` file from the `target` directory, it appears to be fixed. Thank you very much for the assistance!

**Comment by obi1kenobi** [comment]
No worries, glad we got to the bottom of it!
